File: apps/reminder/src/main/java/it/ioapp/com/reminder/model/Message.java
package it.ioapp.com.reminder.model;

import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import it.ioapp.com.reminder.util.Constants;

import org.springframework.data.annotation.Id;

import dto.FeatureLevelType;
import dto.MessageContentType;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class Message {

	@Id
	protected String id;
	protected String senderServiceId;
	protected String senderUserId;
	protected int timeToLiveSeconds;
	protected long createdAt;
	protected boolean isPending = true;
	protected String content_subject = "undefined";
	protected MessageContentType content_type;
	protected double content_paymentData_amount;
	protected String content_paymentData_noticeNumber;
	protected boolean content_paymentData_invalidAfterDueDate;
	protected String content_paymentData_payeeFiscalCode;
	protected String fiscalCode;
	protected String shard = "0";

	@JsonDeserialize(using = LocalDateTimeDeserializer.class)
	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = Constants.DATE_FORMAT_DESERIALIZER)
	protected LocalDateTime dueDate;

	protected FeatureLevelType feature_level_type;
}

File: apps/cqrs-func/src/types/avro/message.ts
import { BaseAvroRecord } from "./BaseAvroRecord";
import { FeatureLevelType } from "./FeatureLevelTypeEnum";
import { MessageContentType } from "./MessageContentTypeEnum";

export interface MessageInterface {
  content_paymentData_amount: number;
  content_paymentData_invalidAfterDueDate: boolean;
  content_paymentData_noticeNumber: string;
  content_paymentData_payeeFiscalCode: string;
  content_subject: string;
  content_type?: MessageContentType | null;
  createdAt: number;
  dueDate: number;
  feature_level_type?: FeatureLevelType | null;
  fiscalCode: string;
  id: string;
  isPending: boolean;
  senderServiceId: string;
  senderUserId: string;
  timeToLiveSeconds: number;
}

export class Message extends BaseAvroRecord implements MessageInterface {
  public static readonly schema: object = {
    doc: "Kafka JS schema for cosmos api container 'messages'",
    fields: [
      {
        default: "undefined",
        name: "fiscalCode",
        type: "string",
      },
      {
        default: "undefined",
        name: "id",
        type: "string",
      },
      {
        default: "undefined",
        name: "senderServiceId",
        type: "string",
      },
      {
        default: "undefined",
        name: "senderUserId",
        type: "string",
      },
      {
        default: 3600,
        name: "timeToLiveSeconds",
        type: "int",
      },
      {
        default: 0,
        logicalType: "time-millis",
        name: "createdAt",
        type: "long",
      },
      {
        default: true,
        name: "isPending",
        type: "boolean",
      },
      {
        default: "undefined",
        name: "content_subject",
        type: "string",
      },
      {
        default: null,
        name: "feature_level_type",
        type: [
          "null",
          {
            name: "FeatureLevelType",
            symbols: ["STANDARD", "ADVANCED"],
            type: "enum",
          },
        ],
      },
      {
        default: null,
        name: "content_type",
        type: [
          "null",
          {
            name: "MessageContentType",
            symbols: ["GENERIC", "PAYMENT", "EU_COVID_CERT", "LEGAL", "PN"],
            type: "enum",
          },
        ],
      },
      {
        default: 0,
        name: "content_paymentData_amount",
        type: "int",
      },
      {
        default: "undefined",
        name: "content_paymentData_noticeNumber",
        type: "string",
      },
      {
        default: false,
        name: "content_paymentData_invalidAfterDueDate",
        type: "boolean",
      },
      {
        default: "undefined",
        name: "content_paymentData_payeeFiscalCode",
        type: "string",
      },
      {
        default: 0,
        logicalType: "time-millis",
        name: "dueDate",
        type: "long",
      },
    ],
    name: "message",
    namespace: "dto",
    type: "record",
  };
  public static readonly subject: string = "message";
  public content_paymentData_amount = 0;
  public content_paymentData_invalidAfterDueDate = false;
  public content_paymentData_noticeNumber = "undefined";
  public content_paymentData_payeeFiscalCode = "undefined";
  public content_subject = "undefined";
  public content_type?: MessageContentType | null = null;
  public createdAt = 0;
  public dueDate = 0;
  public feature_level_type?: FeatureLevelType | null = null;
  public fiscalCode = "undefined";
  public id = "undefined";
  public isPending = true;
  public senderServiceId = "undefined";

  public senderUserId = "undefined";
  public timeToLiveSeconds = 3600;

  public schema(): object {
    return Message.schema;
  }

  public subject(): string {
    return Message.subject;
  }
}
